### PR TITLE
[android][expo-av] Add volume panning for MediaPlayer androidImplemen…

### PR DESCRIPTION
**Pull request by DominickVale**
# Why

Volume panning is a really useful functionality that should be supported in expo-av.

# How

Adding volume panning to the expo-av MediaPlayer implementation should be quite easy since it's already provided by the native API ( https://developer.android.com/reference/android/media/MediaPlayer#setVolume(float,%20float) ). 

The new pan value ranges between -1.0f (full left) and 1.0f (full right) just like iOS AVAudioPlayer's pan ( https://developer.apple.com/documentation/avfaudio/avaudioplayer/1390884-pan )

 For simplicity and clarity sake, i decided to "hardcode/couple" the panning logic to only the MediaPlayer's `updateVolumeMuteAndDuck` method until the panning functionality will be added to SimpleExoPlayer (i'm not 100% confident on this decision)...

On the javascript side, i added the new `pan: number` property to be set when calling `setVolumeAsync(volume, pan)`. I initially wanted to create an additional method `setPan(value)` but considering the way the native API is implemented on Android, i decided it could've been cleaner by just passing it as an optional parameter (again, not 100% confident though)

# Test Plan

Added a few simple tests in the test-suite project.
Added a pan slider in the bare-expo -> APIs -> Audio playground.
I also patched the expo-av build inside one of my personal projects and tested it there.

# Notes
<sup>My second pr! :) I will greatly appreciate any type of feedback.</sup>
I will update the documentation after receiving positive feedback.
Unfortunately, i do not have the ability to work on iOS at the moment.

### CODE REVIEW

File: apps/native-component-list/src/screens/AV/AudioPlayer.tsx [hunk 1]
```diff
@@ -74,7 +76,10 @@ export default class AudioPlayer extends React.Component<Props, State> {
   _loadSoundAsync = async (source: PlaybackSource) => {
     const soundObject = new Audio.Sound();
     try {
-      await soundObject.loadAsync(source, { progressUpdateIntervalMillis: 150 });
+      await soundObject.loadAsync(source, {
+        progressUpdateIntervalMillis: 150,
+        androidImplementation: 'MediaPlayer',
```
Comment (DominickVale) [inline apps/native-component-list/src/screens/AV/AudioPlayer.tsx:81]: Should i attempt to make this toggleable instead of just removing it?

Comment (bbarthec) [reply apps/native-component-list/src/screens/AV/AudioPlayer.tsx:81]: Yeah, let's add the ability to switch the implementation then 👍

**Review by bbarthec — COMMENTED** [review]
Hello @DominickVale 👋 
Thank you for creating this PR and coming up with this `audio panning` idea 😁 
The approach you're proposing seems good to me 👍 
As this feature would be available in non-default android implementation (we default to `ExoPlayer`) we need to document it pretty well 🗒️ 
Nevertheless, in order to fully accept the idea I'll need you to create a plan to deliver this feature for other platforms as well (iOS & Web).
I image you can create a followup PR for other platform (or at least very descriptive issues requesting for implementation on other platforms).

**Comment by DominickVale** [comment]
> in order to fully accept the idea I'll need you to create a plan to deliver this feature for other platforms as well (iOS & Web). I image you can create a followup PR for other platform (or at least very descriptive issues requesting for implementation on other platforms).

@bbarthec Hi! Thank you for considering my pr and reviewing it. I'm currently trying to figure out how to implement this on web, but i think it's going to be a bit more complicated than the MediaPlayer and iOS solution (which, unfortunately, at this moment i don't think i can compile or test since i don't have any apple devices), so, yeah I'll probably just create these two issues.

If just creating the issues are enough for now, i will create them and if you think they are ok then I'll proceed to finish up this pr by applying your suggestions and writing documentation. (If this sounds ok with  you)

**Comment by bbarthec** [comment]
Yeah, I believe it might be good to incrementally work on this feature (platform by platform), so creating issues and polishing this PR sounds good to me 👍

**Comment by DominickVale** [comment]
I made the same mistake again... I don't know why this keeps happening... I'm sorry, how do i remove the review requests?

**Comment by DominickVale** [comment]
I created a discussion instead of an issue for web, i also already wrote an example commit of how it could be added.
https://github.com/expo/expo/discussions/15617

**Comment by DominickVale** [comment]
I also created the ios discussion. https://github.com/expo/expo/discussions/15642 ...
Please let me know if you think the information is enough in both of them or if i should add more details!

**Comment by barthap** [comment]
Hello, I'm sorry that this wasn't reviewed for so long 🙄  - I somehow missed this one as it got lost in tons of GH notifications.

This PR generally looks good to me 👍 There are conflicts now but they should be easy to fix. The only change since then is that documentation is now generated from comments in `*.ts` files, and then the code needs to be rebuilt with `yarn build`.
Are you going to resolve this yourself? Or alternatively, you could give me write permissions to the branch/fork so I fix it myself.

Another thing is missing changelog entry - see _expo-bot_'s comment above.

**Comment by DominickVale** [comment]
> Are you going to resolve this yourself? Or alternatively, you could give me write permissions to the branch/fork so I fix it myself.
> 
> Another thing is missing changelog entry - see _expo-bot_'s comment above.

Hello, no problem at all. I know you guys are very busy. And sure, i will try to fix this tomorrow 😀

**Comment by DominickVale** [comment]
Everything should be ok now... Not sure why the `Test Suite / android (31)` check failed with these two
```
✕ passes Random (76142 ms)
✕ passes Permissions (18963 ms)
```

File: packages/expo-av/src/AV.types.ts [hunk 2]
```diff
@@ -245,6 +249,11 @@ export type AVPlaybackStatusToSet = {
    * > Note that this only affect the audio of this `playbackObject` and do NOT affect the system volume.
    */
   isMuted?: boolean;
+  /**
+   * The current audio panning value of the audio for this media.
+   * > Note that this only affect the audio of this `playbackObject` and do NOT affect the system volume.
+   */
+  audioPan?: number;
```
Comment (barthap) [inline packages/expo-av/src/AV.types.ts:256]: For user-settable properties, you can mark them as platform-specific 😉 
```suggestion
  /**
   * The current audio panning value of the audio for this media.
   * > Note that this only affect the audio of this `playbackObject` and do NOT affect the system volume.
   * @platform android
   */
  audioPan?: number;
```

File: packages/expo-av/src/Video.types.ts [hunk 3]
```diff
@@ -215,6 +215,11 @@ export type VideoProps = {
    * See the [AV documentation](./av) for more information.
    */
   isMuted?: boolean;
+  /**
+   * The desired audio panning value of the audio for this media. This value must be between `-1.0` (full left) and `1.0` (full right).
+   * See the [AV documentation](./av) for more information.
+   */
+  audioPan?: number;
```
Comment (barthap) [inline packages/expo-av/src/Video.types.ts:222]: Does the volume pan work on video too? :D

**Review by barthap — APPROVED** [review]
Mostly looks good to me 😁 I'll try it out myself and merge within a few days

File: packages/expo-av/src/Video.types.ts (hunk 3 above)
Comment (DominickVale) [reply packages/expo-av/src/Video.types.ts:222]: > Does the volume pan work on video too? :D

It does, of course, only with the MediaPlayer implementation

**Pull request merged**